### ISSUE TITLE
How to contribute?

### ISSUE THREAD

**Issue by alivepool** [ProtonMail/ios-mail#1]
How can I contribute to this project? Can you please update the contribution documentation.

**Comment by ssa3512** [comment]
I suspect contribution through github will not be allowed, as the code is licensed GPL which is not allowed in the app store.

**Comment by djoate** [comment]
> I suspect contribution through github will not be allowed, as the code is licensed GPL which is not allowed in the app store.

How does this work? Signal is in the [app store](https://apps.apple.com/us/app/signal-private-messenger/id874139669) and [licensed under GPLv3](https://github.com/signalapp/Signal-iOS/blob/master/LICENSE).

**Comment by abjurato** [comment]
Hi and thank you for attention! At the moment we’ve got a huge refactor in progress so it will be a pity if someone’s contribution will be affected by those changes by the time we’’ll be ready to merge them into main branch. Also, we need to clarify risen licensing questions with our legal team.

We are discussing this topic internally and hope will come up with updates next week

**Comment by mackuba** [comment]
@ssa3512 it's possible to set up a project in such a way that any contributors simply agree to let the owner use the code they submit under any license

**Comment by mcgroarty** [comment]
> @ssa3512 it's possible to set up a project in such a way that any contributors simply agree to let the owner use the code they submit under any license

Yes, this is a very common practice. A company can dual license software, offering binaries under a proprietary license, and source under a license such as GPL or LGPL. Third parties who redistribute the software (or derivatives) have to provide source, while the original owner is not subject to these restrictions.

Contributors would have to assign ownership of their patches to the original project owner in order to support this dual licensing, and this would be enforced by some form of contributor agreement. Or contributors can fork the project and carry on doing only what is permitted by the GPL license without submitting back upstream.

**Comment by diejmon** [comment]
Any updates on this?

**Comment by Ge0rges** [comment]
Any updates? I'd like to contribute.
